File: swagger/models/get_staff_request.py
from .base_model import BaseModel


class GetStaffRequest(BaseModel):
	"""
	no description available

	Attributes:

		staff_ids
		A list of the requested staff IDs.

		filters
		Filters to apply to the search. Possible values are:  * StaffViewable
		* AppointmentInstructor  * ClassInstructor  * Male  * Female

		session_type_id
		Return only staff members that are available for the specified session
		type. You must supply a valid `StartDateTime` and `LocationID` to use
		this parameter.

		start_date_time
		Return only staff members that are available at the specified date and
		time. You must supply a valid `SessionTypeID` and `LocationID` to use
		this parameter.

		location_id
		Return only staff members that are available at the specified
		location. You must supply a valid `SessionTypeID` and `StartDateTime`
		to use this parameter.

		limit
		Number of results to include, defaults to 100

		offset
		Page offset, defaults to 0.
	"""

	input_map = {
		'StaffIds': 'staff_ids',
		'Filters': 'filters',
		'SessionTypeId': 'session_type_id',
		'StartDateTime': 'start_date_time',
		'LocationId': 'location_id',
		'Limit': 'limit',
		'Offset': 'offset',
		}

	output_map = {
		'staff_ids': 'StaffIds',
		'filters': 'Filters',
		'session_type_id': 'SessionTypeId',
		'start_date_time': 'StartDateTime',
		'location_id': 'LocationId',
		'limit': 'Limit',
		'offset': 'Offset',
		}

	staff_ids = None
	filters = None
	session_type_id = None
	start_date_time = None
	location_id = None
	limit = None
	offset = None

File: swagger/Models/GetStaffRequest.php
<?php

namespace App\Mindbody\Models;

/**
 * no description available
 */
class GetStaffRequest extends BaseModel
{
	/**
	 * A list of the requested staff IDs.
	 * 
	 * @var integer[] 
	 */
	public $staffIds;
	/**
	 * Filters to apply to the search. Possible values are:  * StaffViewable
	 * * AppointmentInstructor  * ClassInstructor  * Male  * Female
	 * 
	 * @var string[] 
	 */
	public $filters;
	/**
	 * Return only staff members that are available for the specified session
	 * type. You must supply a valid `StartDateTime` and `LocationID` to use
	 * this parameter.
	 * 
	 * @var integer format:int32
	 */
	public $sessionTypeId;
	/**
	 * Return only staff members that are available at the specified date and
	 * time. You must supply a valid `SessionTypeID` and `LocationID` to use
	 * this parameter.
	 * 
	 * @var string format:date-time
	 */
	public $startDateTime;
	/**
	 * Return only staff members that are available at the specified
	 * location. You must supply a valid `SessionTypeID` and `StartDateTime`
	 * to use this parameter.
	 * 
	 * @var integer format:int32
	 */
	public $locationId;
	/**
	 * Number of results to include, defaults to 100
	 * 
	 * @var integer format:int32
	 */
	public $limit;
	/**
	 * Page offset, defaults to 0.
	 * 
	 * @var integer format:int32
	 */
	public $offset;

	protected function getInputMap()
	{
		return [
			'StaffIds' => ['staffIds', 'array', null],
			'Filters' => ['filters', 'array', null],
			'SessionTypeId' => ['sessionTypeId', 'integer', 'int32'],
			'StartDateTime' => ['startDateTime', 'string', 'date-time'],
			'LocationId' => ['locationId', 'integer', 'int32'],
			'Limit' => ['limit', 'integer', 'int32'],
			'Offset' => ['offset', 'integer', 'int32'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'staffIds' => ['StaffIds', 'array', null],
			'filters' => ['Filters', 'array', null],
			'sessionTypeId' => ['SessionTypeId', 'integer', 'int32'],
			'startDateTime' => ['StartDateTime', 'string', 'date-time'],
			'locationId' => ['LocationId', 'integer', 'int32'],
			'limit' => ['Limit', 'integer', 'int32'],
			'offset' => ['Offset', 'integer', 'int32'],
			];
	}
}
